benchmarks/graphics_pipeline/GraphicsBenchmarkApp.cpp

```diff
@@ -544,15 +535,26 @@ void GraphicsBenchmarkApp::SetupSkyBoxMeshes()
 
 void GraphicsBenchmarkApp::SetupSphereMeshes()
 {
-    OrderedGrid grid(kMaxSphereInstanceCount, kSeed);
+    const uint32_t requiredSphereCount = std::max<uint32_t>(pSphereInstanceCount->GetValue(), kDefaultSphereInstanceCount);
+    const uint32_t initSphereCount     = std::min<uint32_t>(kMaxSphereInstanceCount, 2 * std::max(requiredSphereCount, mInitializedSpheres));
+
+    OrderedGrid grid(initSphereCount, kSeed);
+    mInitializedSpheres = initSphereCount;
+
+    // Destroy the meshes if they were created.
```

Comment:
This triggers a Vulkan validation error:
The destroy can be called while the buffer is still used by the previous frame.
We might want to do a WaitForIdle (quick solution, don't need perf there) or something more fine grained to prevent those issues.

Reply:
Done.